### ISSUE TITLE
fix N+1 query in Analytics V2

### ISSUE THREAD

**Issue by donrestarone** [restarone/violet_rails#1399]
**Describe the bug**

Visiting Analytics V2 causes N+1 query. On markedrestaurant.com the N+1 causes a timeout when the range is 3 months or above: 

<img width="1728" alt="Screen Shot 2023-02-03 at 7 36 50 PM" src="https://user-images.githubusercontent.com/35935196/216735330-29b53c45-0ab0-4343-943c-f7fa9829899c.png">

**Comment by donrestarone** [comment]
In addition to the N+1, there maybe an issue with Analytics V2 that causes a memory leak ⚠️ (evidence WIP)

**Comment by alis-khadka** [comment]
> In addition to the N+1, there maybe an issue with Analytics V2 that causes a memory leak ⚠️ (evidence WIP)

@donrestarone 
Is there anything remaining to be updated here regarding the "memory leak" part?

**Comment by donrestarone** [comment]
@alis-khadka not really, except that you can observe on `top` / `htop` the memory does not get freed after visiting the analytics page. eg; on a fresh boot the app can be using 600MB RAM, and after a user visits the analytics page it will jump to 1.2GB and stay there
